core/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -288,55 +281,75 @@ public boolean isSuggestOnly() {
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         searchType = SearchType.fromId(in.readByte());
-
         indices = new String[in.readVInt()];
         for (int i = 0; i < indices.length; i++) {
             indices[i] = in.readString();
         }
-
         routing = in.readOptionalString();
         preference = in.readOptionalString();
-
         if (in.readBoolean()) {
```

Comment:
Scroll constructor is private, we could make it public, but that's odd...although it's what we have all over the place in the java api.